### PR TITLE
extention vscode start

### PR DESCRIPTION
**Pull request by mbarekh**

### CODE REVIEW

File: src/configurationParser.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+export const modelTags: { [key: string]: string } = {
```
Comment (nicolaserny) [inline src/configurationParser.ts]: Bad naming of this file.
I don't think it's a parser.
Only contains constants

File: src/drawComponent.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
+
+import * as go from "gojs";
+import { Parser } from "parser";
+import { stateMachineColor } from "graphicColors";
+import { NodeDataArrayTemplate, LinkDataArrayTemplate } from "gojsTemplates";
+
+export class DrawComponent {
+
+    private $: any;
+    public diagram: go.Diagram;
```
Comment (nicolaserny) [inline src/drawComponent.ts]: Avoid exposing member as public. Use getter/setter (if required)

File: src/drawComponent.ts [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+    constructor() {
```
Comment (nicolaserny) [inline src/drawComponent.ts]: remove empty constructor

File: src/drawComponent.ts [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+            "nodeDataArray": nodeDataArray,
+            "linkDataArray": linkDataArray
+        };
+        return go.Model.fromJson(data);
+    }
+}
+
+
+
+
```
Comment (nicolaserny) [inline src/drawComponent.ts]: Remove empty lines

File: src/drawComponent.ts [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+export class DrawComponent {
+
+    private $: any;
+    public diagram: go.Diagram;
+
+    constructor() {
+    }
+
+    draw(parser: Parser, divId: string): void {
+        this.$ = go.GraphObject.make;
```
Comment (nicolaserny) [inline src/drawComponent.ts]: Question: why do you need this naming ($) ?

File: src/drawComponent.ts [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+                        new go.Binding("text", "text")),
+                    $(go.Placeholder), { padding: 5 }
+                )
+            );
+        return groupTemplate;
+    }
+
+
+    private getLinkTemplate(): go.Link {
+        const $ = this.$;
```
Comment (nicolaserny) [inline src/drawComponent.ts]: Why do you need this line ?

File: src/extension.ts [hunk 7]
```diff
@@ -0,0 +1,106 @@
+import * as vscode from "vscode";
+import * as path from "path";
+import * as fs from "fs-extra";
+import { ComponentGraphicalModel } from "parserObjects";
+import { TextDocument } from "vscode";
+
+class TextDocumentContentProvider implements vscode.TextDocumentContentProvider {
+	public constructor(private context: vscode.ExtensionContext, private componentGraphicalModel: ComponentGraphicalModel) {
```
Comment (nicolaserny) [inline src/extension.ts]: unnecessary constructor

File: src/extension.ts [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+	}
+	public provideTextDocumentContent(uri: vscode.Uri): string {
+		const container = `<div id="diagram" style="
```
Comment (nicolaserny) [inline src/extension.ts]: It's better to extract html code in a dedicated file.

File: src/extension.ts [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+		const index = this.componentsUri.indexOf(e.uri.toString());
+		if (index > -1) {
+			this.componentsUri.splice(index, 1);
+		}
+	}
+
+	draw() {
+		const modelFile = vscode.window.activeTextEditor.document.fileName;
+		if (/.cxml$/.test(modelFile)) {
+			const tmp = modelFile.split("\\");
```
Comment (nicolaserny) [inline src/extension.ts]: Find a generic path separator.
Your code is unlikely to work on linux

File: src/extension.ts [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+			const componentView = (tmp[tmp.length - 1]).replace(".cxml", "");
+			const uri = `${componentView}://authority/${componentView}`;
+			const index = this.componentsUri.indexOf(uri);
+			if (index > -1) {
+				return;
+			}
+			const graphicalFile = modelFile.replace(".cxml", "") + "_Graphical.xml";
```
Comment (nicolaserny) [inline src/extension.ts]: Should extract some code in a separate method / class

File: src/index.ts [hunk 11]
```diff
@@ -0,0 +1,11 @@
+import { Parser } from "parser";
```
Comment (nicolaserny) [inline src/index.ts]: index.ts, is it the entry point of the viewer or the extension ?
I'm a bit confused by your file structure

File: src/parser.ts [hunk 12]
```diff
@@ -0,0 +1,353 @@
...
+            .getElementsByTagName(graphicalTags.TransitionGraphicalData);
+        this.controlPointTransition = this.getControlPointTransition(transitionGraphicalData);
+
+        transitionGraphicalData = xmlGraphicalDom.getElementsByTagName(graphicalTags.TransversalLinks)[0]
+            .getElementsByTagName(graphicalTags.TransitionGraphicalData);
+        this.controlPointTriggerable = this.getControlPointTransition(transitionGraphicalData);
+    }
+
+    private parseModel(): void {
+        const scxmlDom = (new DOMParser()).parseFromString(this.componentGraphicalModel.model, "text/xml");
```
Comment (nicolaserny) [inline src/parser.ts]: Why using DOMParser ? and not using xml2js like our js api.
It's easier to maintain (instead of doing manual parsing)

**Comment by mbarekh** [comment]
test this please

**Comment by nicolaserny** [comment]
test this please

**Comment by nicolaserny** [comment]
test this please

File: src/provider.ts [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    get onDidChange(): vscode.Event<vscode.Uri> {
+        return this._onDidChange.event;
+    }
+
+    public update(uri: vscode.Uri) {
+        if (!this._waiting) {
+            this._waiting = true;
+            setTimeout(() => {
```
Comment (nicolaserny) [inline src/provider.ts]: useless setTimeout

**Pull request merged**